### ISSUE TITLE
Python 3.11 ModuleNotFoundError: No module named 'tomllib'

### ISSUE THREAD

**Issue by PerchunPak** [psf/black#2983]
**Describe the bug**

In Python 3.11 it is raising exception with `ModuleNotFoundError`.

**To Reproduce**

Install Python 3.11

```sh
$ black .
```

The resulting error is:

```py
Traceback (most recent call last):
  File "/home/runner/work/autodonate/autodonate/.venv/bin/black", line 5, in <module>
    from black import patched_main
    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/runner/work/autodonate/autodonate/.venv/lib/python3.11/site-packages/black/__init__.py", line 52, in <module>
    from black.files import (
    ^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/runner/work/autodonate/autodonate/.venv/lib/python3.11/site-packages/black/files.py", line 25, in <module>
    import tomllib
    ^^^^^^^^^^^^^^
ModuleNotFoundError: No module named 'tomllib'
```

**Environment**

- Black's version: 22.3.0
- OS and Python version: Github Actions Ubuntu, Python 3.11

**Additional context**

[Logs from my CI.](https://github.com/fire-squad/autodonate/runs/5764464577?check_suite_focus=true#step:7:18)
Also, why you haven't did any CI on 3.11?

**Comment by SamuelMarks** [comment]
Also this did work on 3.11.0-alpha.5 just not 3.11.0-alpha.6

**Comment by SamuelMarks** [comment]
Also this did work on 3.11.0-alpha.5 with black-22.1.0-py3-none-any.whl just not 3.11.0-alpha.6 with black-22.3.0-py3-none-any.whl.

**Comment by SamuelMarks** [comment]
Also this did work on `3.11.0-alpha.5` with `black-22.1.0-py3-none-any.whl` just not `3.11.0-alpha.6` with `black-22.3.0-py3-none-any.whl`.

**Comment by pawamoy** [comment]
Would it be possible to change to a try/except instead of a condition on the Python version, at least until tomllib is available in Python 3.11 (alpha or not)? I understand 3.11 is still alpha, but I believe many users want to test their code and run their CI tools (i.e. Black) on Python 3.11. Of course I can pin Black to `<22.3` for Python 3.11 only (with markers), but that's a bit cumbersome to do in every project :confused:

**Comment by JelleZijlstra** [comment]
@pawamoy that's fair, opened #2987
